### PR TITLE
Make completion service return checkpoints

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
The new table for #4681 and the query used to retrieve completions
currently does not return checkpoints. These do not have to match
the application_id and submitting_party query since those fields
are not populated.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
I didn't spot that some of the nullable columns are related when you created the schema. Can I suggest a constraint like the following?

```sql
ALTER TABLE participant_command_completions
  ADD CONSTRAINT participant_command_completions_nullable_columns CHECK (
    -- checkpoints
    (application_id IS NULL AND submitting_party IS NULL AND command_id IS NULL AND transaction_id IS NULL AND status_code IS NULL AND status_message IS NULL)
    -- accepted commands
    OR (application_id IS NOT NULL AND submitting_party IS NOT NULL AND command_id IS NOT NULL AND transaction_id IS NOT NULL AND status_code IS NULL AND status_message IS NULL)
    -- rejected commands
    OR (application_id IS NOT NULL AND submitting_party IS NOT NULL AND command_id IS NOT NULL AND transaction_id IS NULL AND status_code IS NOT NULL AND status_message IS NOT NULL)
  );
```

Verbose as hell, but I think it might be useful.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala [hunk 1]
```diff
@@ -61,8 +61,10 @@ object CommandCompletionsTable {
           from participant_command_completions
           where
             completion_offset between $startInclusive and $endExclusive and
-            application_id = ${applicationId: String} and
-            submitting_party in (${parties.asInstanceOf[Set[String]]})"""
+            (application_id is null or application_id = ${applicationId: String}) and
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala]: Hm, this spreads out the clause for checkpoints across two terms. How about something along the lines of
```sql
-- checkpoints
(application_id is null and submitting_party is null and command_id is null) 
or
-- rejections or successful completions
(application_id = ${applicationId: String} and submitting_party in (${parties}))
```

**Comment by stefanobaghino-da** [comment]
> I didn't spot that some of the nullable columns are related when you created the schema. Can I suggest a constraint like the following?
> 
> ```sql
> ALTER TABLE participant_command_completions
>   ADD CONSTRAINT participant_command_completions_nullable_columns CHECK (
>     -- checkpoints
>     (application_id IS NULL AND submitting_party IS NULL AND command_id IS NULL AND transaction_id IS NULL AND status_code IS NULL AND status_message IS NULL)
>     -- accepted commands
>     OR (application_id IS NOT NULL AND submitting_party IS NOT NULL AND command_id IS NOT NULL AND transaction_id IS NOT NULL AND status_code IS NULL AND status_message IS NULL)
>     -- rejected commands
>     OR (application_id IS NOT NULL AND submitting_party IS NOT NULL AND command_id IS NOT NULL AND transaction_id IS NULL AND status_code IS NOT NULL AND status_message IS NOT NULL)
>   );
> ```
> 
> Verbose as hell, but I think it might be useful.

In principle it's something I would consider but I think we should talk about before adding constraints. Verbosity is not a problem, I'm mostly worried about the failure mode.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala]: Sure, make sense (btw now GitHub supports multi-line suggestions, go wild!)

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala]: I'll stick to `application_id` and `submitting_party` for checkpoints though, `command_id` is not indexed.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala]: Addressed by https://github.com/digital-asset/daml/pull/4735/commits/390025ca382a58b34a71246869dcd1b95fb2be05

**Pull request merged**